### PR TITLE
Feature/omp spec

### PR DESCRIPTION
**Pull request by cangumeli**
I will resolve the conflicts before the merge.

### CODE REVIEW

File: src/spec_kernels.cpp [hunk 1]
```diff
@@ -0,0 +1,339 @@
+#include "../include/spec_kernels.hpp"
+
+// Declare loops
+#define PARALLEL_FOR_2D                                   \
```
Comment (fbaru-dev) [inline src/spec_kernels.cpp:4]: Please add here a header with a list of gates which currently can use the specialized implementation 2

**Review by fbaru-dev — COMMENTED** [review]
According to my review, after addressing the comments, we can accept the pull request

File: include/gate_spec.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  CHadamard=0,
+  CRotationX, CRotationY, CRotationZ,
+  CPauliX, CPauliY, CPauliZ,
+  CPhase,
+  None
+};
+
+}
+
+#endif
```
[the remark this answers is not in the document]
Comment (cangumeli) [reply include/gate_spec.hpp]: Done

File: include/qureg.hpp [hunk 3]
```diff
@@ -219,14 +220,27 @@ class QubitRegister
...
+                              std::size_t sstate_ind, std::size_t estate_ind,
+                              // Spec parameters
+			      qhipster::GateSpec1Q spec=qhipster::GateSpec1Q::None,
+                              BaseType angle=0);
+
+  void Apply1QubitGate(unsigned qubit, TM2x2<Type> const&m,
+                       // Spec parameters 
+		       qhipster::GateSpec1Q spec=qhipster::GateSpec1Q::None,
+                       BaseType angle=0);
+
```
[the remark this answers is not in the document]
Comment (cangumeli) [reply include/qureg.hpp]: I added some inline comments.

File: src/spec_kernels.cpp (hunk 1 above)
Comment (cangumeli) [reply src/spec_kernels.cpp:4]: Currently, it is in the documentation of `TurnOnSpecializeV2` and available in the Doxygen. You can see `qureg_init.cpp` for the documentation.

**Review by fbaru-dev — APPROVED** [review]
The review has been done and we can accept the pull request

**Pull request merged**